test/test_dht_storage.cpp

```diff
@@ -76,11 +76,10 @@ namespace
 
 	bool g_storage_constructor_invoked = false;
 
-	dht_storage_interface* dht_custom_storage_constructor(sha1_hash const& id
-		, dht_settings const& settings)
+	dht_storage_interface* dht_custom_storage_constructor(dht_settings const& settings)
 	{
 		g_storage_constructor_invoked = true;
-		return dht_default_storage_constructor(id, settings);
+		return dht_default_storage_constructor(settings);
```

Comment:
a part of me also feel a bit uneasy about being able to construct a storage without any node IDs. this feels like a two-stage initialization.